README.md

```diff
@@ -1,2 +1,12 @@
 # comp110-worksheet-C
 Base repository for COMP110 worksheet C
+
+(a) The algorithm basically checks through the list to identify any duplications that may be present within the entries. If there are any, it will return as True. Then if not, it'll return as False.
+(b) If the list is in quadratic, it will need to run through the list either twice or multiple times depending on how many items are in the list. Adding more to the list will increase the amount of time taken. 
+(c) The algorithm is still correct, mainly because it still runs the list to see if there are any duplicates.
+(d) It makes it run through the list only once as it is checking each of the positions. Thus making it faster.  
+(e) The time complexity is still in quadratic, this is due to the fact that it goes through the list of i only once now. It's still a linear search within a linear search.
+(f) Information present from the scource [1] suggest that the sort funtion in Python is called 'Timesort' and it utilises the Big O notation. For example, the best method to use is O(n).
```

Comment:
True, however we are usually interested in worst-case time complexity -- which in the case of Timsort, is O(n log n).